### DISCUSSION TITLE
Manage an array of unsorted objects.

### DISCUSSION THREAD

**Discussion by rrirower** [rokucommunity/brighterscript#1273] [Q&A] [answered]
I'm looking to search an roArray of unsorted roAssociativeArray items.  The search is based upon a string entered by the user.  At the present, I perform a linear search.  I'm hoping there's a BrighterScript function or method to perform a faster and more efficient search.

**Comment by TwitchBronBron** [comment, marked as answer]
Unfortunately no, we haven't included any sort functions in brighterscript. I think those types of enhancements would be better served as third party libraries instead of being embedded in brighterscript, so that way you can upgrade the library independently from brighterscript. 

rodash has as [sortBy](https://tkss-software.github.io/rodash/module-rodash.html#.sortBy) function, but looking at the source code it seems to be linear as well. 

This might be a good opportunity to implement a few of the more popular sorting algorithms, see which one performs best for your needs, and if you're feeling motivated, it could even be released as a ropm module or shareable roku library.

**Reply by rrirower** [reply]
Thanks.  Looks like I'll be digging into this further.
